src/controllers/legacy/cmc.utils.ts

```diff
@@ -26,101 +29,78 @@ export async function prepareSummary() {
-      avg: avg(sql`${prices.value}::numeric`).as('avg'),
-      high: max(sql`${prices.value}::numeric`).as('high'),
-      low: min(sql`${prices.value}::numeric`).as('low'),
-    })
-    .from(prices)
-    .where(
-      and(
-        gte(prices.tickAt, dayjs().subtract(1, 'week').toDate()),
-        or(...tokenPairs.map((x) => and(eq(prices.baseId, x.baseId), eq(prices.quoteId, x.quoteId)))),
-      ),
-    )
-    .groupBy(prices.baseId, prices.quoteId, sql`date`)
-    .orderBy(desc(sql`date`))
-    .execute();
+  const priceItems = await getPricesInRange(dayjs().subtract(1, 'week').toDate(), new Date());
 
   const items = pools.map((pool) => {
-    const basePrices = priceItems.filter((item) => item.baseId === pool.base.id && item.quoteId === pool.stablecoin.id);
-    const quotePrices = priceItems.filter(
-      (item) => item.baseId === pool.quote.id && item.quoteId === pool.stablecoin.id,
-    );
+    const basePrices = priceItems.filter((item) => item.tokenId === pool.base.id);
+    const quotePrices = priceItems.filter((item) => item.tokenId === pool.quote.id);
+
+    const basePrices24h = basePrices.filter((x) => dayjs(x.tickAt).unix() >= dayAgo);
+    const quotePrices24h = quotePrices.filter((x) => dayjs(x.tickAt).unix() >= dayAgo);
 
-    const basePrices24h = basePrices.filter((x) => dayjs(x.date).unix() >= dayAgo);
-    const quotePrices24h = quotePrices.filter((x) => dayjs(x.date).unix() >= dayAgo);
+    const lastUsdPrice = basePrices[0]?.avg || '0';
+    const lastPriceQuote = quotePrices[0]?.avg || '0';
+    const lastPrice = lastUsdPrice && lastPriceQuote ? prettyNumber(bignumber(lastUsdPrice).div(lastPriceQuote)) : '0';
 
-    const lastUsdPrice = basePrices[0]?.avg || null;
-    const lastPriceQuote = quotePrices[0]?.avg || null;
-    const lastPrice = lastUsdPrice && lastPriceQuote ? bignumber(lastUsdPrice).div(lastPriceQuote).toFixed(18) : null;
-    const baseDayPrice = basePrices.find((x) => dayjs(x.date).unix() <= dayAgo)?.avg || null;
-    const baseWeekPrice = basePrices[basePrices.length - 1]?.avg || null;
+    const baseDayPrice = basePrices.find((x) => dayjs(x.tickAt).unix() <= dayAgo)?.avg || lastUsdPrice;
+    const baseWeekPrice = basePrices[basePrices.length - 1]?.avg || lastUsdPrice;
 
-    const quoteDayPrice = quotePrices.find((x) => dayjs(x.date).unix() <= dayAgo)?.avg || null;
-    const quoteWeekPrice = quotePrices[quotePrices.length - 1]?.avg || null;
+    const quoteDayPrice = quotePrices.find((x) => dayjs(x.tickAt).unix() <= dayAgo)?.avg || lastPriceQuote;
+    const quoteWeekPrice = quotePrices[quotePrices.length - 1]?.avg || '0';
 
-    const baseHigh24 = basePrices24h.map((x) => bignumber(x.high));
-    const baseHigh25Usd = baseHigh24.length ? bnMax(baseHigh24).toFixed(18) : null;
+    const baseHigh24 =
+      basePrices24h.length > 0 ? basePrices24h.map((x) => bignumber(x.high)) : [bignumber(lastPriceQuote)];
+    const baseHigh25Usd = baseHigh24.length ? prettyNumber(bnMax(baseHigh24)) : '0';
```

Comment:
```suggestion
    const baseHigh24Usd = baseHigh24.length ? prettyNumber(bnMax(baseHigh24)) : '0';
```
This is probably just a typo.